### PR TITLE
Remove blank lines

### PR DESCRIPTION
**Pull request by TadeasKucera**
This feature removes double blank lines and use just one blank line instead when there is too many blank lines in the rule (Iff more than one third of the new-lines in the rule is followed by another new-line). The removing is only performed inside of rules.

### CODE REVIEW

File: src/types/token_stream.cpp [hunk 1]
```diff
@@ -291,6 +294,39 @@ void TokenStream::determineNewlineSectors()
...
+		{
+			++lineCounter;
+			if (wasLine)
+				++doubleLineCounter;
+			wasLine = true;
+		}
+		else
+			wasLine = false;
+	}
+	return 3 * doubleLineCounter > lineCounter;
```
Comment (TadeasKucera) [inline src/types/token_stream.cpp]: This means that when more than half of the newlines is "doubled", then all "doubled" newlines are made simple new line.

Comment (metthal) [reply src/types/token_stream.cpp]: This kind of comment would better fit directly in the source code :) Thanks.

**Comment by metthal** [comment]
LGTM however I have a question. Does it only cover double or even triple, quadruple, .... blank lines?

**Comment by TadeasKucera** [comment]
Thank you very much for noticing, this only covers double blank lines, because naturally the `std::list<T,Allocator>::erase` function returns iterator **after** the last removed element. Fixed in last commit.

**Pull request merged**